src/taco-dependency-installer/installers/androidPackagesInstaller.ts

```diff
@@ -0,0 +1,296 @@
+            } else {
+                deferred.reject(err);
+            }
+        });
+        cp.on("exit", (code: number) => {
+            if (errorOutput) {
+                this.telemetry
+                    .add("error.description", "ErrorOnExitOfChildProcess on installAndroidPackages", /*isPii*/ false)
+                    .add("error.code", code, /*isPii*/ false)
+                    .add("error.message", errorOutput, /*isPii*/ true);
+                deferred.reject(new Error(errorOutput));
+            } else {
+                this.availablePackages = [];
+
+                // Parse for package ids and save them
+                var regex: RegExp = /id: \d+ or "(.+)"/g;
+                var match: RegExpMatchArray = regex.exec(output);
+
+                while (match) {
+                    this.availablePackages.push(match[1]);  // Index 0 is the entire matched line, index 1 is the captured group which is the actual package ID
+                    match = regex.exec(output);
+                }
+
+                deferred.resolve({});
+            }
+        });
+
+        return deferred.promise;
+    }
+
+    private getPlatformToolsPackageId(): string {
+        // Look in the list of available packages for the platform-tools package
+        var packageIdIndex: number = this.availablePackages.indexOf(AndroidPackagesInstaller.PLATFORM_TOOLS_PKG);
+
+        // If none is found, it means the latest platform tools are already installed, so just return an empty string; otherwise return the full ID
+        return packageIdIndex === -1 ? "" : this.availablePackages[packageIdIndex];
+    }
+
+    private getBuildToolsPackageId(): string {
+        // Look in the list of available packages for one that starts with the build-tools prefix
+        var packageId: string = "";
+
+        this.availablePackages.some(function (pkg: string): boolean {
+            if (pkg.indexOf(AndroidPackagesInstaller.BUILD_TOOLS_PKG_PREFIX) === 0) {
+                packageId = pkg;
+
+                return true;
+            }
+
+            return false;
+        });
+
+        // If none is found, it means the latest build tools are already installed, so just return an empty string; otherwise return the full ID
+        return packageId;
+    }
+
+    private getAndroidTargetPackageId(): string {
+        // Read the Android project properties
+        var projectPropertiesPath: string = path.join("platforms", "android", "project.properties");
+        var fileContent: string = fs.readFileSync(projectPropertiesPath).toString();
+
+        // Find the value at the "target=" line
+        var androidTargetId: string = /target=(.+)/.exec(fileContent)[1];
+
+        // Make sure the android target is in the list of available packages, otherwise it is an error state
+        if (this.availablePackages.indexOf(androidTargetId) === -1) {
+            throw new Error(resources.getString("UnknownAndroidTarget", androidTargetId, projectPropertiesPath));
```

Comment:
These are not really TACOxxxx errors; we're just making some of the low-level installer errors clearer, so I would prefer to not have enum entries in TacoErrorCodes for them